lib/collection/src/collection/collection_ops.rs

```diff
@@ -193,14 +201,19 @@ impl Collection {
 
             replica_set.remove_peer(peer_id).await?;
 
-            // Collect shard transfers related to removed shard...
-            let transfers = shard_holder
-                .get_transfers(|transfer| transfer.from == peer_id || transfer.to == peer_id);
+            let all_nodes_drop_transfers = self
+                .channel_service
+                .all_peers_at_version(ABORT_TRANSFERS_ON_SHARD_DROP_FROM_VERSION);
+            if all_nodes_drop_transfers {
```

Comment:
I've added a simple version gate here. Only drop ongoing transfers to a shard we delete if all nodes are at least 1.9.

It makes the logic more robust, which is worth it in my opinion.

Though it's unlikely to actually go wrong, it'll be painful if it happens. That is because we don't have any instruments to fix such problems.

@generall I heard you're not a fan of version gating here. Is that also true in this specific use case?

Reply:
I think I imagined version-gating through the request to other peers for some reason. But this version looks good to me, it is guaranteed by consensus that during execution of this operation all nodes will see the same versions of other nodes